builtin/providers/aws/data_source_aws_db_snapshot_test.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSDbSnapshotDataSource_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAwsDbSnapshotDataSourceConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsDbSnapshotDataSourceID("data.aws_db_snapshot.snapshot"),
```

Comment:
Can we add a second step that adds another snapshot, and ensure that our data source then points to that newest snapshot? The test as it stands doesn't test the `most_recent` logic, because there is only 1 snapshot. If possible we should compare the ARNs and/or check that the "most recent" data source ARN changes